modules/elasticsearch/src/main/java/org/testcontainers/elasticsearch/ElasticsearchContainer.java

```diff
@@ -101,10 +104,26 @@ public ElasticsearchContainer(final DockerImageName dockerImageName) {
     @Override
     protected void containerIsStarted(InspectContainerResponse containerInfo) {
         if (isAtLeastMajorVersion8) {
-            byte[] bytes = copyFileFromContainer("/usr/share/elasticsearch/config/certs/http_ca.crt", IOUtils::toByteArray);
+            extractCert("/usr/share/elasticsearch/config/certs/http_ca.crt");
+        }
+    }
+
+    /**
+     * Reads CA cert from a certain path in the container
+     *
+     * @param path Path of the CA cert
+     */
+    public void extractCert(String path) {
```

Comment:
This public method creating a side effect on a started container feels a bit un-idiomatic to the general Testcontainers API to me. 

WDYT about something like this:
* We make `extractCert` private and have it been called in `containerIsStarted`, as we do now
* In addition we add a setter for container creation, `withCertPath` that simply sets this value to be used  by `containerIsStarted` later.

Reply:
That clearly is a much better idea than my 10pm coding skills! Will fix

Reply:
done